### ISSUE TITLE
[workspaces] Assertion and weird output

### ISSUE THREAD

**Issue by theodelrieu** [conan-io/conan#4752]
Hello,

I'm using the new workspaces feature in 1.13.1 on a side project, and an assertion gets raised:

```
File "/Users/theo/Library/Python/3.6/lib/python/site-packages/conans/client/output.py", line 119, in write
    assert self.scope != "virtual", "printing with scope==virtual"
AssertionError: printing with scope==virtual
```

Additionally, there are a lot of redundant build requirement outputs like the following:

```
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_cmake/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: mgs_test_helpers/0.1.0@mgs/testing
mgs_meta/0.1.0@mgs/testing: Applying build-requirement: catch2/2.2.2@bincrafters/stable
```

Steps to reproduce:

* `git clone https://github.com/theodelrieu/playground --branch workspaces`
* `cd playground && mkdir build && cd build && conan workspace install ../conanws.yml`

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by lasote** [comment]
Please @memsharded take a look. I managed to reproduce it easily but looks far from being obvious what is the issue or issues here. Something with the build requires looks very messed up and I really don't understand the "virtual" node assert.

**Comment by memsharded** [comment]
Working on it, confirmed it is a bug, and it is an extremely complicated one. Will update as soon as possible.

**Comment by theodelrieu** [comment]
Confirmed fixed with the planned fix for 1.13.2

**Comment by lasote** [comment]
1.13.2 Released!
